### PR TITLE
controllers: set controller watches on owned configmaps & statefulsets

### PR DESCRIPTION
**Pull request by phlogistonjohn**
When a configmap or statefulset is created by the operator in response to an SmbShare resource, we set the SmbShare as the owner (controller owner). We want our reconcile loop to be called when a change is made to one of these resources.

### CODE REVIEW

File: controllers/smbshare_controller.go [hunk 1]
```diff
@@ -83,6 +83,8 @@ func (r *SmbShareReconciler) SetupWithManager(mgr ctrl.Manager) error {
 	return ctrl.NewControllerManagedBy(mgr).
 		For(&sambaoperatorv1alpha1.SmbShare{}).
 		Owns(&corev1.PersistentVolumeClaim{}).
+		Owns(&corev1.ConfigMap{}).
 		Owns(&appsv1.Deployment{}).
+		Owns(&appsv1.StatefulSet{}).
```
Comment (anoopcs9) [inline controllers/smbshare_controller.go:88]: I don't remember the discussion outcome but what about `corev1.Service`?

Comment (phlogistonjohn) [reply controllers/smbshare_controller.go:88]: Sure, anything resource type w generate from an SmbShare is fair game (although not required).

**Review by anoopcs9 — APPROVED** [review]
lgtm, thanks.

**Review by synarete — APPROVED** [review]
LGTM

**Pull request merged**